src/NuGet.Core/NuGet.Configuration/Settings/ISettingsFile.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace NuGet.Configuration
+{
+    /// <summary>
+    /// Interface to expose a NuGet settings file
+    /// </summary>
+    internal interface ISettingsFile
+    {
+        /// <summary>
+        /// Folder under which the config file is present
+        /// </summary>
+        string Root { get; }
+
+        /// <summary>
+        /// The file name of the config file. Joining <see cref="Root"/> and
+        /// <see cref="FileName"/> results in the full path to the config file.
+        /// </summary>
+        string FileName { get; }
+
+        /// <summary>
+        /// Full path to the settings file
+        /// </summary>
+        string ConfigFilePath { get; }
+
+        /// <summary>
+        /// Defines if the settings file is considered a machine wide settings file
+        /// </summary>
+        /// <remarks>Machine wide settings files cannot be eddited.</remarks>
+        bool IsMachineWide { get; }
+
+        /// <summary>
+        /// Saves the document to disk
+        /// </summary>
+        void Save();
+
+        /// <summary>
+        /// Root element for the settings abstraction
+        /// </summary>
+        NuGetConfiguration RootElement { get; }
```

Comment:
I think handling all the manipulation from this class is limiting. At the end of the day, the user of this interface knows how the data is stored and how to handle it. If we add specific methods for interacting with the config files here we would be constrained to these APIs (the same way we were constraint by the previous APIs in `ISettings`). This is the reason I opted for exposing the root element and letting the user handle the data directly. Also, note that this is an internal API, therefore the user should know how the data works. 

Any non-internal user should always interact with the computed settings through `ISettings` instead of `ISettingsFile`.

Reply:
`If we add specific methods for interacting with the config files here we would be constrained to these APIs (the same way we were constraint by the previous APIs in ISettings).`

Not really, what i'm asking is to have apis similar to `ISettings` to manage sections which isn't constraint by current data.

`This is the reason I opted for exposing the root element and letting the user handle the data directly`

This is the part I dont understand, if we have already defined an interface to interact with individual setting file then why would I expose an element to directly handle the underneath data?

`Also, note that this is an internal API, therefore the user should know how the data works.`

This has nothing to do with either internal or public as long as there are multiple layers and each layer knows what it's suppose to do. In current code, It feels like there are multiple layers but functionality and responsibilities are not distributed among them... Also, if you properly handle all the data manipulation at `ISettingFile` then tomorrow exposing individual config files in UI, will be far more convenient and doable.

Reply:
My initial thought was that adding an API to manage this would not be necessary since the providers that could call directly into this will know how to handle the API, but I can add the following methods to `ISettingsFile` and not expose the root element so that the API is consistent with `ISettings`:

```
SettingsSection GetSection(string sectionName); 

bool CreateSection(SettingsSection section, bool isBatchOperation = false); 

bool SetItemInSection(string sectionName, SettingsItem item, bool isBatchOperation = false); 
```